test/bun.js/jest.test.js

```diff
@@ -0,0 +1,43 @@
+import { beforeAll, it, describe, expect, afterAll } from 'bun:test';
+
+var before = false;
+var after = false;
+
+beforeAll(async () => {
+  if (before) throw new AssertionError("before should be false");
+  if (after) throw new AssertionError("after should be false");
+  await new Promise(resolve => setTimeout(() => resolve(before = true), 1000));
+});
+
+it("should run after beforeAll()", async () => {
+  expect(before).toBe(true);
+  expect(after).toBe(false);
+});
+
+afterAll(() => {
+  expect(before).toBe(true);
+  expect(after).toBe(false);
+  after = true;
+});
+
+describe("describe", async () => {
+  var before = false;
+  var after = false;
+
+  beforeAll(async () => {
+    if (before) throw new AssertionError("before should be false");
+    if (after) throw new AssertionError("after should be false");
+    before = await new Promise(resolve => setTimeout(() => resolve(true), 1000));
```

Comment:
this test will take a second to run

can you make this 1ms instead

Reply:
Even with 500ms this test will pass most of the time :sweat_smile:

Reply:
Since `beforeEach()` fails reliably without `setTimeout()`, and this PR does fix the common logic for both, I've updated the tests such that `beforeAll()` will _not_ fail (most of the time) without this fix, but won't be holding up dead time.